nowcasting_dataloader/datamodules.py

```diff
@@ -0,0 +1,156 @@
+"""Datamodules to use for training"""
+import logging
+import os
+from typing import List, Optional, Tuple, Union
+
+import torch
+from nowcasting_dataset.config import load_yaml_configuration
+from nowcasting_dataset.config.model import Configuration
+from pytorch_lightning import LightningDataModule
+
+from nowcasting_dataloader.datasets import SatFlowDataset, worker_init_fn
+
+_LOG = logging.getLogger(__name__)
+_LOG.setLevel(logging.DEBUG)
+
+
+class SatFlowDataModule(LightningDataModule):
+    """
+    Example of LightningDataModule for SatFlow dataset.
+
+    A DataModule implements 5 key methods:
+        - prepare_data (things to do on 1 GPU/TPU, not on every GPU/TPU in distributed mode)
+        - setup (things to do on every accelerator in distributed mode)
+        - train_dataloader (the training dataloader)
+        - val_dataloader (the validation dataloader(s))
+        - test_dataloader (the test dataloader(s))
+    This allows you to share a full dataset without explaining how to download,
+    split, transform and process the data.
+    Read the docs:
+        https://pytorch-lightning.readthedocs.io/en/latest/extensions/datamodules.html
+    """
+
+    def __init__(
+        self,
+        temp_path: str,
+        configuration: Union[Configuration, str],
+        n_train_data: int = 24900,
```

Comment:
we can now get this from the configruation

Reply:
Ah okay, cool